packages/waku/src/lib/builder/build.ts

```diff
@@ -293,6 +298,24 @@ const buildServerBundle = async (
   if (!('output' in serverBuildOutput)) {
     throw new Error('Unexpected vite server build output');
   }
+
+  if (serve === 'cloudflare') {
+    // For Cloudflare Pages, create `_routes.json` which prevents calls to
+    // static client assets from invoking the server-side function.
+    // TODO: check for existing `_routes.json` in users' root directory? And
+    // merge this with default staticRoutes provided below
+    const routesJsonFilePath = joinPath(rootDir,
+      config.distDir,
+      '_routes.json');
+    const staticRoutes = {
+      version: 1,
+      include: ['/*'],
+      exclude: ['/public/assets/*']
+    };
+
+    await writeFile(routesJsonFilePath, JSON.stringify(staticRoutes));
+  }
```

Comment:
I think there's a reason I didn't put it there. Let me check.

And thanks for commenting on this, you made me notice that the public and asset paths should probably be variables instead of hardcoded

Reply:
I was wrong - we absolutely can (and should) move this to `output-cloudflare.ts`. Good spot!